src/main/comments.js

```diff
@@ -387,10 +389,21 @@ function printLeadingComment(commentPath, print, options) {
   // Leading block comments should see if they need to stay on the
   // same line or not.
   if (isBlock) {
-    return concat([
-      contents,
-      hasNewline(options.originalText, options.locEnd(comment)) ? hardline : " "
-    ]);
+    const leftNode =
```

Comment:
Can we probably just use `comment.precedingNode` here instead of `leftNode`?

Reply:
How about:
```js
    const lineBreak = hasNewline(options.originalText, options.locEnd(comment))
      ? hasNewline(options.originalText, options.locStart(comment), {
          backwards: true
        })
        ? hardline
        : line
      : " ";
```
?

Reply:
**Input:**
```js
if (foo) /* baz */
  bar();

if (foo)
/* baz */
bar();
```
**Current output:**
```js
if (foo)
  /* baz */
  bar();

if (foo)
  /* baz */
  bar();
```
**Output with my proposed change:**
```js
if (foo) /* baz */ bar();

if (foo)
  /* baz */
  bar();
```

I would say it's okay to do this. If the user wants this comment to be on the next line, they can move it manually, and it will stay there.

Reply:
@thorn0 Thank you! I did not know `backwards` option. Could you tell me what does that mean?

Reply:
It determines the direction, in which `hasNewLine` searches for a new line. By default, it searches the source code forward from the given index. With `backwards: true` it searches backwards.

```hasNewline(options.originalText, options.locStart(comment), { backwards: true })```

This returns `true` if there is a new line between the comment and the token preceding it.

Reply:
Would be good to hear from @evilebottnawi or @j-f1 if they're okay with this change.